### DISCUSSION TITLE
Help with _hc based on filter selected

### DISCUSSION THREAD

**Discussion by Deanobeano2** [cortexrd/Knack-Toolkit-Library#392] [Q&A] [answered]
Hi

I'm wondering if it's possible to hide a column based on the filter selected.

In our case I'd like to hide the "Mark As" column only if the "Completed" filter has been clicked. This is so that staff can't mark an already completed job as completed again.

![image](https://github.com/user-attachments/assets/5be784fa-c045-4cdf-99f4-2fe5fb940f9c)

Dean

**Comment by amandajane-mo** [comment]
Hi Dean,

This isn't currently possible with KTL, to hide a column based on a filter
(as far as I'm aware). You could however harness the ktl function
hideTableColumn and unhideTableColumn, which could be triggered on the
click event of the filter buttons.

Amanda

**Comment by Deanobeano2** [comment]
Hi Amanda

Is this a new keyword, as I don't see it in the Keywords page?

Dean

**Comment by amandajane-mo** [comment]
Hey Dean,

It’s not a keyword but a function in the ktl that you can call from your
code. It’s the function used by the _hc keyword.

Let me if need any guidance.

Amanda

**Comment by Deanobeano2** [comment]
Hi Amanda

I think I'll need guidance with this, if you would be so kind.

Dean

**Comment by cortexrd** [comment]
Dean,

Wouldn't is be simpler to add a display rule to the **Mark As** action column to hide the action when its **Job state** value is **Completed**?

If you can't do it in the Builder, maybe it's because Job state is a connected field?

Then you can certainly use the _cfv keyword with the hide option.

**Comment by Deanobeano2** [comment]
Norm

Thank you.

That worked and sorry for over-looking the obvious.

Dean

**Comment by cortexrd** [comment, marked as answer]
@Deanobeano2 
Glad it worked out.
**Don't forget to set as Answered!**

**Reply by Deanobeano2** [reply]
Thanks
